dali/operators/reader/loader/loader.h

```diff
@@ -146,11 +150,14 @@ class Loader {
 
     int samples_to_choose_from = initial_buffer_fill_;
     if (shards_.front().start == shards_.front().end) {
-      if (!is_new_epoch && pad_last_batch_) {
+      if ((returned_sample_counter_  < num_samples(num_shards_, Size()) || !is_new_epoch) &&
```

Comment:
I don't fully understand the logic behind this. Can you add couple of sentences as a comment here?

Reply:
Done